adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -235,7 +235,23 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     }
   }
 
-  def loadBam(filePath: String): RDD[AlignmentRecord] = {
+  /**
+   * Loads a SAM/BAM file.
+   *
+   * This reads the sequence and record group dictionaries from the SAM/BAM file
+   * header. SAMRecords are read from the file and converted to the
+   * AlignmentRecord schema.
+   *
+   * @param filePath Path to the file on disk.
+   *
+   * @return Returns a Tuple3 containing (an RDD of reads, the sequence
+   *   dictionary representing the contigs these reads are aligned to
+   *   if the reads are aligned, the record group dictionary for the reads
+   *   if one is available).
+   *
+   * @see loadAlignments
+   */
+  def loadBam(filePath: String): (RDD[AlignmentRecord], SequenceDictionary, RecordGroupDictionary) = {
```

Comment:
I'd prefer to funnel users towards loading/saving the tuple. My reasoning is that the tuple is the full set of information you need to reconstruct all the alignment data (the reads themselves + information about the sequencing process that generated the reads + information about the sequences the reads were aligned to). There _are_ places where we don't use all of the tuple, but the cost of loading the second and third elements of the tuple should be small relative to the cost of loading the RDD.

I'd like to understand why you'd like the primary entrypoint to be loading just the RDD. What are the pros/cons from your side?